sched_boundary/sched_boundary.py

```diff
@@ -117,50 +115,34 @@ def register_cbs(self):
         if gcc.get_main_input_filename() in self.mod_srcs | {self.fake}:
             super().register_cbs()
 
-    @staticmethod
-    def add_fn_list(decl, fn_lst):
-        signature = {"fn": decl.name,
-                "ret": GccBugs.fix(decl.result, decl.result.type.str_no_uid),
-                "params": ", ".join(GccBugs.fix(arg, arg.type.str_no_uid) \
-                        for arg in decl.arguments) if decl.arguments else "void"
-            }
-        GccBugs.variadic_function(decl, signature)
-        fn_lst.append([decl,
-            signature,
-            decl.location.line - 1,
-            decl.location.column - 1,
-            decl.function.end.line - 1,
-            decl.function.end.column - 1])
-
-    def function_define(self, decl, _):
-        # only func definition will trigger PLUGIN_FINISH_PARSE_FUNCTION
+    def function_location(self):
         loc = gcc.get_location()
```

Comment:
I suspect whether this works? After moving it to final_work, `gcc.get_location` always returns the same code line despite where they're defined.